lib/Runtime/Provisioner/Provisioner.cpp

```diff
@@ -530,4 +538,24 @@ void Provisioner::cleanupProvision(llvm::ArrayRef<std::string> names,
       functions_.erase(name);
     }
   }
+  if (failure) {
+    // Remove any partitions added to devices.
+    for (auto &device : deviceCleanup) {
+      for (auto network : device.second) {
+        std::promise<void> evictPromise;
+        auto done = evictPromise.get_future();
+        std::unique_ptr<Error> evictErr;
+        devices_[device.first]->evictNetwork(
+            network, [&evictErr, &evictPromise](std::string, Error err) {
+              evictErr = glow::make_unique<Error>(std::move(err));
+              evictPromise.set_value();
+            });
+        done.wait();
```

Comment:
This seems somewhat boilerplate -- and I think we have pretty similar code in HostManager's `removeNetwork()`. Worth colocating this logic in a helper or something?